### ISSUE TITLE
In router boot program, cleanup never fires on SIGINT or SIGTERM

### ISSUE THREAD

**Issue by krancour** [deis/deis#4254]
I'm not sure how important the cleanup that is to be performed is, but I'm pretty sure it's not happening.

I haven't looked yet, but this is likely happening in other spots as well.

https://github.com/deis/deis/blob/master/router/boot.go#L88

```
func main() {
  // ...
  exitChan := make(chan os.Signal, 2)
  signal.Notify(exitChan, syscall.SIGTERM, syscall.SIGINT)
  go cleanOnExit(exitChan)
  <-exitChan
}
```

The issue is that the main goroutine can potentially exit (and does) before the `cleanOnExit()` function performs cleanup.

Here's a trivial program that demonstrates the techniques does not work.

```
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"
)

func main() {
	exitChan := make(chan os.Signal, 2)
	signal.Notify(exitChan, syscall.SIGTERM, syscall.SIGINT)
	go cleanOnExit(exitChan)
	fmt.Println("Waiting for interrupt...")
	<-exitChan
}

func cleanOnExit(exitChan chan os.Signal) {
	for _ = range exitChan {
		fmt.Println("Cleaning up...")
	}
}
```

**Comment by bacongobbler** [comment]
This is okay. This is going to be dead code once I'm done with what I'm currently working on. This should have used a syncgroup or something to make sure that the cleanup function did finish up, but it will all be gone soon :)

**Comment by bacongobbler** [comment]
We should investigate the other components, but I think this is the only component that has a post-cleanup goroutine AFAICR

**Comment by krancour** [comment]
You're probably right, but let's keep this open until we've had a chance to review those others spots.

**Comment by krancour** [comment]
Closed by #4260
